### ISSUE TITLE
false positive on empty_line_after_outer_attr with async-trait

### ISSUE THREAD

**Issue by Freyskeyd** [rust-lang/rust-clippy#5567]
Hello!

I think I found an issue with the `empty_line_after_outer_attr` when using `async-trait` crate.

here's the clippy output

```rs
error: Found an empty line after an outer attribute. Perhaps you forgot to add a `!` to make it an inner attribute?
  --> src/main.rs:19:5
   |
19 |     async fn fetch_user_feed(&self, _user_id: &str) -> Result<(), ClientError> {
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: the lint level is defined here
  --> src/main.rs:2:9
   |
2  | #![deny(clippy::nursery)]
   |         ^^^^^^^^^^^^^^^
   = note: `#[deny(clippy::empty_line_after_outer_attr)]` implied by `#[deny(clippy::nursery)]`
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#empty_line_after_outer_attr

error: aborting due to previous error

error: could not compile `async-trait-bug`.
```

the `cargo clippy -V`

```rs
clippy 0.0.212 (204bb9b5 2020-03-17)
```

and you can find the source here: https://github.com/Freyskeyd/async-trait-bug

Thanks!

**Comment by phansch** [comment]
Managed to reduce this down to:

```rust
#![deny(clippy::empty_line_after_outer_attr)]

use async_trait::async_trait;

#[async_trait]
pub trait FetchUserFeed {
    async fn a() {
        let _a = "";



    }
}
```

When there's at least 3 empty lines in the method body it triggers the lint. I'll try and build a repro without `async_trait` next.

**Comment by phansch** [comment]
Further reduced down to this, without `async-trait`:

```rust
// no-prefer-dynamic

#![crate_type = "proc-macro"]
#![feature(repr128, proc_macro_hygiene, proc_macro_quote)]
#![allow(clippy::identity_conversion)]

extern crate proc_macro;
extern crate quote;
extern crate syn;

use quote::{quote, quote_spanned};
use syn::parse_macro_input;
use syn::{parse_quote, ItemTrait, TraitItem};
use proc_macro::TokenStream;

#[proc_macro_attribute]
pub fn fake_async_trait(_args: TokenStream, input: TokenStream) -> TokenStream {
    let mut item = parse_macro_input!(input as ItemTrait);
    for inner in &mut item.items {
        if let TraitItem::Method(method) = inner {
            let sig = &method.sig;
            let block = &mut method.default;
            if let Some(block) = block {
                let brace = block.brace_token;

                let my_block = quote_spanned!( brace.span => {
                    // Should not trigger `empty_line_after_outer_attr`
                    #[crate_type = "lib"]
                    #sig #block
                    Vec::new()
                });
                *block = parse_quote!(#my_block);
            }
        }
    }
    TokenStream::from(quote!(#item))
}
```
